.evergreen/run-atlas-tests.sh

```diff
@@ -15,27 +16,37 @@ if [ -z "$PYTHON_BINARY" ]; then
 fi
 
 IMPL=$(${PYTHON_BINARY} -c "import platform, sys; sys.stdout.write(platform.python_implementation())")
-if [ $IMPL = "Jython" -o $IMPL = "PyPy" ]; then
-    echo "Using Jython or PyPy"
+
+if [ $IMPL = "Jython" ]; then
+    # The venv created by createvirtualenv is incompatible with Jython
     $PYTHON_BINARY -m virtualenv --never-download --no-wheel atlastest
     . atlastest/bin/activate
-    trap "deactivate; rm -rf atlastest" EXIT HUP
-    pip install certifi
-    PYTHON=python
 else
-    IS_PRE_279=$(${PYTHON_BINARY} -c "import sys; sys.stdout.write('1' if sys.version_info < (2, 7, 9) else '0')")
+    # All other pythons work with createvirtualenv.
+    . .evergreen/utils.sh
+    createvirtualenv $PYTHON_BINARY atlastest
+fi
+trap "deactivate; rm -rf atlastest" EXIT HUP
+
+if [ $IMPL = "Jython" -o $IMPL = "PyPy" ]; then
+    echo "Using Jython or PyPy"
+    python -m pip install certifi
+else
+    IS_PRE_279=$(python -c "import sys; sys.stdout.write('1' if sys.version_info < (2, 7, 9) else '0')")
     if [ $IS_PRE_279 = "1" ]; then
         echo "Using a Pre-2.7.9 CPython"
-        $PYTHON_BINARY -m virtualenv --never-download --no-wheel atlastest
-        . atlastest/bin/activate
-        trap "deactivate; rm -rf atlastest" EXIT HUP
-        pip install pyopenssl>=17.2.0 service_identity>18.1.0
-        PYTHON=python
+        python -m pip install pyopenssl>=17.2.0 service_identity>18.1.0
     else
         echo "Using CPython 2.7.9+"
-        PYTHON=$PYTHON_BINARY
     fi
 fi
 
-echo "Running tests"
-$PYTHON test/atlas/test_connection.py
+echo "Running tests without dnspython"
+python test/atlas/test_connection.py
+
+# dnspython is incompatible with Jython so don't test that combination.
```

Comment:
See my comment in the Jira ticket: https://jira.mongodb.org/browse/PYTHON-1878?focusedCommentId=3548124&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-3548124